JavaComparator/Comparator.java

```diff
@@ -0,0 +1,133 @@
+import java.util.ArrayList;
+
+public class Comparator {
```

Comment:
Wherever we are using string constants like "c" , "f" , extract them out in a static variable in the beginning of the file with the variable name explaining what is the constant.